kv/memberlist/memberlist_client.go

```diff
@@ -955,30 +965,62 @@ func (m *KV) NotifyMsg(msg []byte) {
 		level.Error(m.logger).Log("msg", "failed to decode received value, unknown codec", "codec", kvPair.GetCodec())
 		return
 	}
+	var wCh chan valueUpdate
 
-	// we have a ring update! Let's merge it with our version of the ring for given key
-	mod, version, err := m.mergeBytesValueForKey(kvPair.Key, kvPair.Value, codec)
+	m.workersMu.Lock()
+	wCh = m.workersChannels[kvPair.Key]
+	if wCh == nil {
```

Comment:
`wCh` -> `ch`

Also `ch := m.workersChannels[kvPair.Key]` is fine, no need for `var`.